hicommonwealth/commonwealth
# buildRetryStrategy + CustomRetryStrategyError

## Link to Issue
Closes: #7638

## Description of Changes
- Adds a `buildRetryStrategy` function
- Added types for RetryStrategies
- Added a `CustomRetryStrategyError` class used to define custom retry strategies for edge-cases

## Test Plan
- Ensure this env var is set `ALLOWED_EVENTS=ThreadCreated`
- Start RabbitMQ
- Start the local server
- Start the message relayer with `pnpm workspace commonwealth start-message-relayer`
- Create a thread in any community
  - The message relayer logs should show that a thread created event was successfully published

## Deployment Plan

1. 

## Other Considerations

-
